### PR TITLE
Media Manager with SVG preview

### PR DESCRIPTION
**Pull request by Rike-cz**
Add SVG support in Media Manager (issue #3986, #3106).

In your config/cms.php file set `'storage.media.imageExtensions'` to `['jpg', 'jpeg', 'bmp', 'png', 'webp', 'gif', 'svg']` to force SVG previewing.

### CODE REVIEW

File: modules/backend/widgets/MediaManager.php [hunk 1]
```diff
@@ -1290,72 +1290,81 @@ protected function getPlaceholderId($item)
...
-            $width = $thumbnailInfo['width'];
-            $height = $thumbnailInfo['height'];
-            $lastModified = $thumbnailInfo['lastModified'];
+        if($this->isVector($thumbnailInfo['path'])) {
+            $markup = $this->makePartial('thumbnail-image', [
+                'isError' => false,
+                'imageUrl' => Url::to(config('cms.storage.media.path') . $thumbnailInfo['path'])
+            ]);
+        }
+        else {
```
Comment (w20k) [inline modules/backend/widgets/MediaManager.php]: @Rike-cz, could you fix styling? 😉 ([style guide](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md))

Why didn't you wrap it with `try-catch` block? `makePartial` - method could throw an exception and your temp.

**Comment by w20k** [comment]
@Rike-cz, while testing, couldn't get a sidebar preview for the SVG files.

Screenshot:
screenshot 2018-12-27 at 18 50 35

File: modules/backend/widgets/MediaManager.php (hunk 1 above)
Comment (Rike-cz) [reply modules/backend/widgets/MediaManager.php]: ok, I will repair this.

**Comment by Rike-cz** [comment]
Did you change storage.media.imageExtensions in cms.php? By default is support for handling SVG as image switched off. This is only option.

**Comment by w20k** [comment]
@Rike-cz, my bad. Now it's all good!

File: modules/backend/widgets/mediamanager/assets/less/mediamanager.less [hunk 2]
```diff
@@ -343,6 +343,7 @@ div[data-control="media-manager"] {
 
         img {
             max-width: 100%;
+            max-height: 224px;
```
Comment (w20k) [inline modules/backend/widgets/mediamanager/assets/less/mediamanager.less]: @Rike-cz, why do you need `max-height` in here?

**Comment by Rike-cz** [comment]
Please new tests here! There were frontend bug.

**Comment by w20k** [comment]
@Rike-cz could you resolve conflicts, will re-test and peer review ;) Thanks!

**Comment by LukeTowers** [comment]
Conflicts resolved. @w20k @datune

**Comment by w20k** [comment]
@LukeTowers re-tested works great! Will review again, and merge, if you don't have any comments to add.

File: modules/backend/widgets/MediaManager.php [hunk 3]
```diff
@@ -1296,52 +1311,59 @@ protected function generateThumbnail($thumbnailInfo, $thumbnailParams = null)
...
-             */
             $path = $thumbnailInfo['path'];
-            $width = $thumbnailInfo['width'];
-            $height = $thumbnailInfo['height'];
-            $lastModified = $thumbnailInfo['lastModified'];
 
-            if (!is_numeric($width) || !is_numeric($height) || !is_numeric($lastModified)) {
-                throw new ApplicationException('Invalid input data');
-            }
+            if($this->isVector($path)) {
```
Comment (LukeTowers) [inline modules/backend/widgets/MediaManager.php]: space after if

**Comment by LukeTowers** [comment]
Comment added @w20k. Note that the completed tag only gets added after the fix is merged in.

File: modules/backend/widgets/MediaManager.php (hunk 3 above)
Comment (Rike-cz) [reply modules/backend/widgets/MediaManager.php]: Done!

**Comment by w20k** [comment]
@LukeTowers okey, I was thinking completed is when it's, sort of, ready to be merged with all the requirement meant.

**Comment by Zmove** [comment]
Does the addition to the milestone means it's implemented ? Cause I still don't have preview thumbnail for svg files in the media manager plugin with the latest version of october.

**Comment by LukeTowers** [comment]
@Zmove did you do
> In your config/cms.php file set 'storage.media.imageExtensions' to ['jpg', 'jpeg', 'bmp', 'png', 'webp', 'gif', 'svg'] to force SVG previewing.

**Comment by Zmove** [comment]
@LukeTowers ah no I didn't knew that parameter. Thank you, it perfectly works.

PS : Don't you think it should be available for all new ocms install ? SVG preview support seems quite a basic need in 2020

**Comment by LukeTowers** [comment]
@Zmove svg uploads are disabled by default for security reasons, if developers are careless with them and include them directly in the HTML instead of in an image src attribute then they can expose themselves to XSS attacks.

**Comment by Zmove** [comment]
@LukeTowers Ah ok didn't know that, thank you for your answer.

**Pull request merged**